vlasovsolver/velocity_space_diffusion.cpp

```diff
@@ -0,0 +1,443 @@
+    
+    for (size_t i = 0; i < betaParaArray.size(); i++) {
+        std::getline(FILEDmumu,linenu0);
+        std::istringstream issnu0(linenu0);
+        std::vector<Real> tempLINE;
+        float numTEMP;
+        while((issnu0 >> numTEMP)) { tempLINE.push_back(numTEMP); }
+        for (size_t j = 0; j < tempLINE.size(); j++) {nu0Array[i][j] = tempLINE [j];}
+    }
+
+
+
+    const auto LocalCells=getLocalCells();
+    #pragma omp parallel for private(fcount,fmu,dfdmu,dfdmu2,dfdt_mu)
+    for (size_t CellIdx = 0; CellIdx < LocalCells.size(); CellIdx++) { //Iterate through spatial cell
+
+        phiprof::start("Initialisation");
+        auto CellID                        = LocalCells[CellIdx];
+        SpatialCell& cell                  = *mpiGrid[CellID];
+	const Real* parameters             = cell.get_block_parameters(popID);
+        const vmesh::LocalID* nBlocks      = cell.get_velocity_grid_length(popID);
+        const vmesh::MeshParameters& vMesh = getObjectWrapper().velocityMeshes[0];      
+
+        Real density_pre_adjust  = 0.0;
+        Real density_post_adjust = 0.0;
+
+        for (size_t i=0; i<cell.get_number_of_velocity_blocks(popID)*WID3; ++i) {
+            density_pre_adjust += cell.get_data(popID)[i];
+        }
+
+        Real Sparsity    = 0.01 * cell.getVelocityBlockMinValue(popID);
+
+        Real dtTotalDiff = 0.0; // Diffusion time elapsed
+
+        Real Vmin   = 0.0; // In case we need to avoid center cells
+        Real Vmax   = 2*sqrt(3)*vMesh.meshLimits[1];
+        Real dVbins = (Vmax - Vmin)/nbins_v;  
+    
+        std::array<Real,4> dfdt;
+
+        std::array<Real,3> bulkV = {cell.parameters[CellParams::VX], cell.parameters[CellParams::VY], cell.parameters[CellParams::VZ]};
+        phiprof::stop("Initialisation");
+
+        std::array<Real,3> B = {cell.parameters[CellParams::PERBXVOL] +  cell.parameters[CellParams::BGBXVOL],
+                                 cell.parameters[CellParams::PERBYVOL] +  cell.parameters[CellParams::BGBYVOL],
+	                         cell.parameters[CellParams::PERBZVOL] +  cell.parameters[CellParams::BGBZVOL]};
+
+        Real Bnorm           = sqrt(B[0]*B[0] + B[1]*B[1] + B[2]*B[2]);
+        std::array<Real,3> b = {B[0]/Bnorm, B[1]/Bnorm, B[2]/Bnorm};
+  
+
+        // There is probably a simpler way to do all of the following but I was too lazy to try to understand how to do it in C++ so I wrote everything myself.       
+        // Build Pressure tensor for calculation of anisotropy and beta
+        Real Ptensor[3][3] = {{cell.parameters[CellParams::P_11],cell.parameters[CellParams::P_12],cell.parameters[CellParams::P_13]},
+                               {cell.parameters[CellParams::P_12],cell.parameters[CellParams::P_22],cell.parameters[CellParams::P_23]},
+                               {cell.parameters[CellParams::P_13],cell.parameters[CellParams::P_23],cell.parameters[CellParams::P_33]}};
+
+        // Build Rotation Matrix
```

Comment:
Yeah this looks pretty horrible :D
it would be a lot better to use the Eigen library for this.